src/dkpy/d_scale_fit.py

```diff
@@ -70,55 +71,42 @@ class DScaleFitSlicot(DScaleFit):
     Compute ``mu`` and ``D`` at each frequency and fit a transfer matrix to ``D``
 
     >>> P, n_y, n_u, K = example_skogestad2006_p325
-    >>> block_structure = np.array([[1, 1], [1, 1], [2, 2]])
+    >>> block_list = [
+    ...     dkpy.ComplexFullBlock(1, 1),
+    ...     dkpy.ComplexFullBlock(1, 1),
+    ...     dkpy.ComplexFullBlock(2, 2),
+    ... ]
+    >>> uncertainty_structure = dkpy.UncertaintyBlockStructure(block_list)
     >>> omega = np.logspace(-3, 3, 61)
     >>> N = P.lft(K)
     >>> N_omega = N(1j * omega)
     >>> mu_omega, D_omega, info = dkpy.SsvLmiBisection().compute_ssv(
     ...     N_omega,
-    ...     block_structure,
+    ...     uncertainty_structure,
     ... )
-    >>> D, D_inv = dkpy.DScaleFitSlicot().fit(omega, D_omega, 2, block_structure)
-    >>> print(control.ss2tf(D[0, 0]))
-    <TransferFunction>: sys[3641]$indexed$converted
-    Inputs (1): ['u[0]']
-    Outputs (1): ['y[0]']
-    <BLANKLINE>
-    <BLANKLINE>
-    0.0157 s^2 + 0.257 s + 0.1391
-    -----------------------------
-      s^2 + 0.9658 s + 0.01424
-    <BLANKLINE>
-    >>> print(control.ss2tf(D[1, 1]))
-    <TransferFunction>: sys[3641]$indexed$converted
-    Inputs (1): ['u[1]']
-    Outputs (1): ['y[1]']
-    <BLANKLINE>
-    <BLANKLINE>
-    0.01573 s^2 + 0.2574 s + 0.1394
-    -------------------------------
-       s^2 + 0.9663 s + 0.01425
-    <BLANKLINE>
+    >>> D, D_inv = dkpy.DScaleFitSlicot().fit(omega, D_omega, 2, uncertainty_structure)
     """
 
     def fit(
         self,
         omega: np.ndarray,
         D_omega: np.ndarray,
         order: Union[int, np.ndarray] = 0,
-        block_structure: Optional[np.ndarray] = None,
+        uncertainty_structure: Optional[
+            uncertainty_structure.UncertaintyBlockStructure
+        ] = None,
     ) -> Tuple[control.StateSpace, control.StateSpace]:
         # Get mask
-        if block_structure is None:
+        if uncertainty_structure is None:
             mask = -1 * np.ones((D_omega.shape[0], D_omega.shape[1]), dtype=int)
```

Comment:
I feel like this behaviour should be wrapped into the `UncertaintyBlockStructure` class if possible (like if you instantiate it with `None` in the constructor). But it may not make sense if the dimensions are not available in that class.